test/coverage/BUILD

```diff
@@ -0,0 +1,115 @@
+# This file is generated by test/coverage/gen_build.sh. DO NOT EDIT. Rerun
```

Comment:
Bazel supports tagging, we use it in this patch to opt tests into the monolithic build via gen_build.sh and its bazel query. Unfortunately, dependencies can't make use of tagging information when we assemble a binary, see https://github.com/bazelbuild/bazel/issues/2362.

Here's what I propose to avoid the fragility:

1. We don't check in test/coverage/BUILD at all.
2. We have the `tests/run_envoy_coverage.sh` script invoke `test/coverage/gen_build.sh` prior to the build. The resulting build file is ephemeral and we don't have stale BUILD files in the repository.

Does that sound acceptable?

For ASAN and other testing, I think we should try and avoid a monolithic build. The reason for this is that when we switch to the next generation CI, there will be the ability to treat individual tests targets independently as first-class resources, with the ability to dive into their history and see them on a test grid. If you prefer we put them in a single binary for Travis CI performance until we go there, please let me know and I can make this non-coverage specific.